File: java/add_whitelisted_ips.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.io.OutputStreamWriter;
import java.net.HttpURLConnection;
import java.net.URL;
import java.net.URLConnection;
import java.util.stream.Collectors;

public class add_whitelisted_ips {

	public static void main(String[] args) throws IOException {
		// Details retrieved from Authentication part.
		String userId = "";
		String token = "";
		
		String payload = "{\"IPAddressList\":[\"xx.xx.xx.xx\",\"xx.xx.xx.xx\"]}"; //Addition of only 1 IP is also possible
		
		URL url = new URL("https://api.smartproxy.com/v1/users/"+userId+"/whitelisted-ips");
		
		URLConnection connection = url.openConnection();
		HttpURLConnection httpConn = (HttpURLConnection) connection;
		
		
		httpConn.setRequestProperty("User-Agent", "Mozilla/5.0");
		httpConn.setRequestProperty ("Authorization", "Token "+token);
		httpConn.setRequestMethod("POST");
		httpConn.setRequestProperty("Content-Type", "application/json; charset=UTF-8");
		httpConn.setRequestProperty("Accept", "application/json");

		httpConn.setDoOutput(true);
		OutputStream outStream = httpConn.getOutputStream();
		OutputStreamWriter outStreamWriter = new OutputStreamWriter(outStream, "UTF-8");
		outStreamWriter.write(payload);
		outStreamWriter.flush();
		outStreamWriter.close();
		outStream.close();
		
		if (200 <= httpConn.getResponseCode() && httpConn.getResponseCode() <= 299) {
		    BufferedReader br = new BufferedReader(new InputStreamReader(httpConn.getInputStream()));
			System.out.print(br.lines().collect(Collectors.joining()));
		} else {
		    BufferedReader br = new BufferedReader(new InputStreamReader(httpConn.getErrorStream()));
			System.out.print(br.lines().collect(Collectors.joining()));
		}
		
		httpConn.disconnect();
		
	}
}

File: php/add_whitelisted_ips.php
<?php

// Information obtained from authentication

$userId = "";
$token = "";

$url = 'https://api.smartproxy.com/v1/users/' . $userId . '/whitelisted-ips';

$data = [
    "IPAddressList" => ["", ""] // You can add a single IP aswell
];

$payload = json_encode($data);

$ch = curl_init();

curl_setopt($ch, CURLOPT_URL, $url);
curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "POST");

$headers = array();
$headers[] = "Accept: application/json";
$headers[] = "Content-Type: application/json";
$headers[] = "Authorization: Token $token";
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
curl_setopt($ch, CURLOPT_POSTFIELDS, $payload);

$result = curl_exec($ch);
if (curl_errno($ch)) {
    echo "Error: " . curl_error($ch);
}
curl_close($ch);
echo $result;
